commands/util.go

```diff
@@ -604,16 +604,20 @@ func getChildValueStrings(keyValueArr whisk.KeyValueArr, key string, childKey st
 	return res
 }
 
-func getValueFromJSONResponse(field string, response map[string]interface{}) interface{} {
+func getValueFromJSONResponse(field string, response interface{}) interface{} {
 	var res interface{}
 
-	for key, value := range response {
-		if key == field {
-			res = value
-			break
+	if result, ok := response.(map[string]interface{}); ok {
+		for key, value := range result {
+			if key == field {
+				res = value
+				break
+			}
 		}
 	}
-
+	if result, ok := response.([]interface{}); ok {
```

Comment:
Should we insist that `field` be "" if response is an array?  I'm not sure exactly how this function is used, but it might be error prone to just silently return the array result when the caller asked for a specific field of what they must have been expecting to be a JSON object.

Reply:
@dgrove-oss yes, you right.

But it is better to cherry this pr's commits to https://github.com/apache/openwhisk-cli/pull/517 and close this pr
I know it is not good due to `Fix deprecated issue(pull/517)` and this pr is target is different.

And due to dependency problem, it seems we must do this,
The reason is https://github.com/apache/openwhisk-cli/pull/517#issuecomment-1221219583

Reply:
@ningyougang 
This method was used to print the invocation result.
https://github.com/apache/openwhisk-cli/blob/92c251f5d3a873029ccd8d815409582301d1103d/commands/action.go#L238

Do we need to print the array result itself for the array case?

Reply:
I tested in such case (btw, i stopped all invokers containers)
![image](https://user-images.githubusercontent.com/11749867/185824239-edb27481-f5c6-4b84-a5ce-430aa916352c.png)

The result is nil
![image](https://user-images.githubusercontent.com/11749867/185824343-ff2fef7a-79ab-46de-aeaf-5b5d3f4d3cd9.png)

Finally is
![image](https://user-images.githubusercontent.com/11749867/185824375-d0ac4bd0-bb2e-4c99-b694-58fc181b9033.png)

If i added `sleep 61s` for that action and make all invokers up again, the result is
![image](https://user-images.githubusercontent.com/11749867/185824455-e44b3419-30aa-488a-b66c-3879f8dd7e32.png)

For normally hello-array action with this pr, the result is
![image](https://user-images.githubusercontent.com/11749867/185824674-d916ac2c-5b61-4c3b-adf7-29372395ac47.png)

Reply:
Btw, i already cherry-pick this pr's commits to https://github.com/apache/openwhisk-cli/pull/517,
Because add-array-result feature is merged before than this pr, so when build this pr and execute this pr's test cases, it would execute openwhisk core relative's add-array-result feature test cases. e.g. when invoke below test case
![image](https://user-images.githubusercontent.com/11749867/185826163-7e317072-d2f5-4ad1-a887-e4d0932e26dc.png)
it would invoke use `CLIActivationsOperations` to get result
![image](https://user-images.githubusercontent.com/11749867/185826242-ee8d7dc8-ca3e-4e4f-afc7-5946ae77b3a9.png)

When build openwhisk pr, it would use `RestActivationsOperations` to get the result

Reply:
ok, it seems now I get it.

Previously, it returns `var res interface{}` without any initialization, when it failed to find the `field`.
Is this an empty string in case it could not find anything?
Isn't it nil?

Reply:
Yes, it would return emtry string now if it failed to find the `field`

Reply:
Was it an empty string or nil?

Reply:
nil